app/models/request/statemachine.rb

```diff
@@ -192,7 +192,13 @@ def on_blocked; end
   def on_hold; end
 
   def failed_upstream!
-    fail_from_upstream! unless failed?
+    # Don't transition it again if it's already failed
+    return if failed?
```

Comment:
Might be worth including 'cancelled?' in here. The state-machine (https://github.com/aasm/aasms) also provides methods like can_fail_from_upstream? which might be worth considering instead. Trade off there is between making the code less likely to fail, vs. not catching unexpected scenarios. (although given the fail_from_upstream! transition is still explicit, this seems reasonable)